From ad91d66c570ff152d0716894ccd58dc6039648d3 Mon Sep 17 00:00:00 2001
From: David Daney <ddaney@caviumnetworks.com>
Date: Wed, 27 Apr 2011 16:39:28 -0700
Subject: [PATCH 010/236] MIPS: Invalidate old TLB mappings when updating huge page PTEs.

Without this, stale Icache or TLB entries may be used.

Signed-off-by: David Daney <ddaney@caviumnetworks.com>
To: linux-mips@linux-mips.org
https://patchwork.linux-mips.org/patch/2318/
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit ca3a9ec9d2f69d1636cda00c5c6cb84a09a96ebd in
git://git.linux-mips.org/pub/scm/linux)

Signed-off-by: Phil Staub <Phil.Staub@windriver.com>
---
 arch/mips/include/asm/hugetlb.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/mips/include/asm/hugetlb.h b/arch/mips/include/asm/hugetlb.h
index e11b46c..ed5059b 100644
--- a/arch/mips/include/asm/hugetlb.h
+++ b/arch/mips/include/asm/hugetlb.h
@@ -71,6 +71,7 @@ static inline pte_t huge_ptep_get_and_clear(struct mm_struct *mm,
 static inline void huge_ptep_clear_flush(struct vm_area_struct *vma,
 					 unsigned long addr, pte_t *ptep)
 {
+	flush_tlb_mm(vma->vm_mm);
 }
 
 static inline int huge_pte_none(pte_t pte)
-- 
1.7.0

